oid-info.md

```diff
@@ -0,0 +1,28 @@
+# Rekor OID Information
+
+## Description
+
+This document defines Rekor
+[OID values](https://github.com/sigstore/sigstore/blob/main/docs/oid-info.md).
+
+Rekor reserves the `1.3.6.1.4.1.57264.3` OID root for all of its values.
+
+## Directory
+
+| OID                       | Name                   | Description                                                                                                   |
```

Comment:
drive-by: would it make sense to use a single OID for a "sigstore bundle" that is using the spec at protobuf-specs?

(That would avoid a whole mess of comments that we've sorted out there: e.g. what encoding is the Log ID, is Verification required, what encoding is the root hash, is the log index the global, etc)

Reply:
My only concern is there would be duplicated data that may be in other locations (body/attestation) - for gitsign body is the main concern, but as long as it's a hashed checksum I guess it wouldn't be that much more?

Let me try playing around with this in my draft PR and see how much of a pain it is.

Reply:
To confirm, do you mean to turn this into the sigstore bundle OID instead of Rekor?

I think what's here should be transferable to the protobuf-specs spec? Though I think some of the naming isn't aligned.

Reply:
>  but as long as it's a hashed checksum I guess it wouldn't be that much more?

Yes, if it's not an attestation it would just be the checksum of the body.

Yes! I was suggesting turning it into a sigstore bundle OID

Reply:
> What would be better is if there was a modified version of [TransparencyLogEntry](https://github.com/sigstore/protobuf-specs/blob/c8a23a43dc1c685d336d112864ba237a15061c53/protos/sigstore_rekor.proto#LL89C9-L89C29), but let the client re-construct the canonicalized_body from other data it already has.

Good point! if you had the TransparencyLogEntry with a nil canonicalized_body, that seems to be all the info needed for you to verify the Rekor signature for offline verification, right?

Reply:
I think so. If we go down this route, we should modify in the protobuf spec to make it clear this is an expected behavior / usage of TransparencyLogEntry. Do either of these options sound better to you?

1. We break out canonicalized_body from TransparencyLogEntry and create a new wrapper message that combines them.
2. We document the canonicalized_body field to say that it's valid for this field to be unset, in which case it's the responsibility of the client to construct it.

FWIW, it looks like canonicalized_body wasn't in the original version of TransparencyLogEntry - https://github.com/sigstore/protobuf-specs/pull/10. We don't have the same problem with Gitsign w.r.t. DSSE canonicalization, so being able to deterministically reconstruct the body shouldn't be an issue.

Reply:
I guess if @haydentherapper is OK with adding Rekor specific OID, that's fine - if something changes with the semantics then I worry this won't be maintained, but if GitSign is one of the only consumers that's OK

Reply:
If there's a way to do this for the protos, definitely on board.

Another option is to move these OIDs back to Gitsign and not have this in Rekor, if Gitsign really is the only client.

Reply:
The more I think about it, maybe that is the best way to proceed. GitSign uses `.3`, defines its own child OIDs, we concurrently define OIDs for the bundle, and then at some point, GitSign adopts the bundle and these new OIDs.